polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -219,6 +321,59 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
     );
   }
 
+  private setCellRef = ({
+    ref,
+    index,
+    inStickyHeader,
+  }: {
+    ref: HTMLTableCellElement | null;
+    index: number;
+    inStickyHeader: boolean;
+  }) => {
+    if (ref == null) {
+      return;
+    }
+
+    if (inStickyHeader) {
+      this.stickyHeadings[index] = ref;
+    } else {
+      this.tableHeadings[index] = ref;
+      this.tableHeadingWidths[index] = ref.offsetWidth;
```

Comment:
Use `ref.getBoundingClientRect().width;`. It will give you a `float` vs `number` and prevents the slight jump that currently occurs when it switches from the actual to the sticky header.